reconcile/dynatrace_token_provider/ocm.py

```diff
@@ -75,19 +97,23 @@ class OCMClient:
     Thin OOP wrapper around OCMBaseClient to avoid function mocking in tests
     """
 
-    def __init__(self, ocm_client: OCMBaseClient):
+    def __init__(self, ocm_client: OCMBaseClient, url: str):
         self._ocm_client = ocm_client
+        self._url = url
```

Comment:
is this `_url` added only for error handling? Can you get url from `HTTPError` object? the error object should be able to get url by `e.request.url`, then no need to wrap `OCMAPIError` to all methods.

Reply:
it's ok if once OCM env stable we revert all changes in this pr, if we want to make it as permanent change, I would not add new dependencies to `OCMClient` only for this error handling.
To make stage and integration endpoint check more strict instead of just string `in` check, we can use `urlparse(url).hostname` to do exactly match.

Reply:
OCM env in stage/int will never be regarded as stable. However, DTPv3 will change a lot here, since currently the error handling in this integration is quite messy.

For this case, I believe it is simpler and safer to just add the extra attr, instead of none checks and additional function calls in catch block. But this will change overall in DTPv3 refactor :+1:

Reply:
now that follow https://github.com/app-sre/qontract-reconcile/pull/4866#discussion_r1955877199, no env url needed to check

Reply:
I still want to add the endpoint (host) as metric label, so we can distinguish severity

Reply:
which metric?

Reply:
Mentioned here https://github.com/app-sre/qontract-reconcile/pull/4866#discussion_r1955877199

Reply:
then let's put `url` related changes in metrics PR, only include parts needed to deal with 500.